### ISSUE TITLE
oead cannot be installed pip, but shown via search

### ISSUE THREAD

**Issue by zardini123** [zeldamods/oead#6]
Running ```pip3 install oead``` results in the error below.  Running ```pip3 search oead``` shows that the package exists.  According to a [user on stackoverflow](https://stackoverflow.com/a/9578018) this issue is solved via the package author themselves.

```
ERROR: Could not find a version that satisfies the requirement oead (from versions: none)
ERROR: No matching distribution found for oead
```

**Comment by leoetlino** [comment]
No binary packages are provided for macOS, so you'll need to build oead and the Python module from source. You'll need to clone this repository, install CMake 3.12+, a compiler that supports C++17, Python `wheel` and then run `pip3 install -e .` at the root of the repo.

**Comment by zardini123** [comment]
I did notice that this repo uses Git modules, so I had to initialize them first using ```git submodule update --init --recursive``` as compiling threw an error stating a CMake file is missing.  After doing that, I tried running the install command again and got this error:
```
    -- c4core: using C++ standard: C++11
    -- c4core: setting global C++ standard: 11
    CMake Error at ../lib/rapidyaml/ext/c4core/cmake/c4Project.cmake:1777 (message):
      not implemented
    Call Stack (most recent call first):
      ../lib/rapidyaml/ext/c4core/CMakeLists.txt:75 (c4_install_exports)
    
    
    -- Configuring incomplete, errors occurred!
```
Do you have any idea what this means?  Because I don't :(

**Comment by leoetlino** [comment]
Uh oh, it looks like rapidyaml (one of the dependencies) doesn't support macOS:

https://github.com/biojppm/cmake/blob/65ad68963bb12810480b24a5c2a36cb347b77145/c4Project.cmake#L1842-L1843

But that line doesn't seem to be useful, so maybe try to remove it (`message(FATAL_ERROR "not implemented")`) and see if it builds?

**Comment by zardini123** [comment]
Removed it and got errors during compiling:
```
oead/lib/rapidyaml/ext/c4core/src/c4/memory_resource.cpp:68:5: error: pasting formed '"NOT IMPLEMENTED: ""need to implement an aligned allocation for this platform"', an invalid preprocessing token
        C4_NOT_IMPLEMENTED_MSG("need to implement an aligned allocation for this platform")
```
Also were given many, many warnings.

Is this the end of the road for macOS users?  Event-editor uses oead and states to be compatible with mac, but if one can't install oead on Mac, one cannot install event-editor on Mac.  :(

**Comment by leoetlino** [comment]
Could you remove these lines and try again?

https://github.com/biojppm/c4core/blob/e054eaf2b6a7344aee9e4e11ac58f81f3d0a8800/src/c4/memory_resource.cpp#L62-L63

Apparently macOS does support aligned allocation so this should be simple to fix

**Comment by zardini123** [comment]
Haha keep removing checks until something works, I like your style!

But it fails again:
```
<oead_path>/oead/lib/rapidyaml/src/c4/yml/tree.cpp:1453:15: error: no matching function for call to 'from_chars'
        if( ! from_chars(tk, &idx))
```

Here's the full error:

```
ERROR: Command errored out with exit status 1:
 command: /usr/local/opt/python@3.8/bin/python3.8 -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'<oead_path>/oead/setup.py'"'"'; __file__='"'"'<oead_path>/oead/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' develop --no-deps
     cwd: <oead_path>/oead/
Complete output (232 lines):
running develop
running egg_info
writing oead.egg-info/PKG-INFO
writing dependency_links to oead.egg-info/dependency_links.txt
writing top-level names to oead.egg-info/top_level.txt
reading manifest file 'oead.egg-info/SOURCES.txt'
reading manifest template 'MANIFEST.in'
writing manifest file 'oead.egg-info/SOURCES.txt'
running build_ext
-- pybind11 v2.4.dev4
-- ryml: using C++ standard: C++11
-- ryml: setting global C++ standard: 11
-- ryml: importing subproject c4core (SUBDIRECTORY)... <oead_path>/oead/lib/rapidyaml/ext/c4core
-- c4core: using C++ standard: C++11
-- c4core: setting global C++ standard: 11
-- ryml: -----> target ryml PUBLIC incorporating lib c4core
-- Using CMake version 3.18.1
-- ZLIB_HEADER_VERSION: 1.2.11
-- ZLIBNG_HEADER_VERSION: 1.9.9
-- Arch detected: 'x86_64'
-- Basearch of 'x86_64' has been detected as: 'x86'
-- Architecture supports unaligned reads
-- Architecture-specific source files: arch/x86/x86.c;arch/x86/insert_string_sse.c;arch/x86/deflate_quick.c;arch/x86/fill_window_sse.c;arch/x86/slide_sse.c;arch/x86/crc_folding.c
-- The following features have been enabled:

 * CMAKE_BUILD_TYPE, Build type: Release (selected)
 * WITH_OPTIM, Build with optimisation
 * WITH_NEW_STRATEGIES, Use new strategies
 * WITH_SSE2, Build with SSE2
 * WITH_SSE4, Build with SSE4
 * WITH_PCLMULQDQ, Build with PCLMULQDQ
 * SSE42_CRC, Support CRC hash generation using the SSE4.2 instruction set, using "-msse4"
 * SSE42_DEFLATE_QUICK, Support SSE4.2-accelerated quick compression
 * PCLMUL_CRC, Support CRC hash generation using PCLMULQDQ, using "-mpclmul"

-- The following REQUIRED packages have been found:

 * PythonInterp
 * PythonLibsNew
 * Threads

-- The following features have been disabled:

 * ZLIB_COMPAT, Provide a zlib-compatible API
 * WITH_GZFILEOP, Compile with support for gzFile-related functions
 * WITH_SANITIZERS, Build with address sanitizer and all supported sanitizers other than memory sanitizer
 * WITH_MSAN, Build with memory sanitizer
 * WITH_FUZZERS, Build test/fuzz
 * WITH_AVX2, Build with AVX2
 * WITH_MAINTAINER_WARNINGS, Build with project maintainer warnings
 * WITH_CODE_COVERAGE, Enable code coverage reporting

-- Configuring done
-- Generating done
-- Build files have been written to: <oead_path>/oead/build/temp.macosx-10.14-x86_64-3.8
[  4%] Built target absl_int128
[  7%] Built target absl_log_severity
[  7%] Built target absl_exponential_biased
[  8%] Built target absl_spinlock_wait
[  9%] Built target absl_dynamic_annotations
[ 15%] Built target oead_res
[ 16%] Built target absl_time_zone
[ 24%] Built target zlib
Scanning dependencies of target ryml
[ 27%] Built target absl_periodic_sampler
[ 27%] Built target absl_strerror
[ 27%] Built target absl_leak_check_disable
[ 29%] Built target absl_leak_check
[ 31%] Built target yaml
[ 32%] Building CXX object oead/lib/rapidyaml/CMakeFiles/ryml.dir/src/c4/yml/tree.cpp.o
[ 32%] Built target absl_city
[ 32%] Built target absl_civil_time
[ 33%] Building CXX object oead/lib/rapidyaml/CMakeFiles/ryml.dir/ext/c4core/src/c4/memory_resource.cpp.o
Scanning dependencies of target c4core
[ 34%] Built target absl_random_seed_gen_exception
[ 35%] Built target absl_random_internal_randen_slow
[ 36%] Built target absl_random_internal_randen_hwaes_impl
[ 37%] Built target absl_raw_logging_internal
[ 37%] Building CXX object oead/lib/rapidyaml/subprojects/c4core/build/CMakeFiles/c4core.dir/src/c4/memory_resource.cpp.o
Scanning dependencies of target absl_strings_internal
[ 37%] Built target absl_random_internal_randen_hwaes
[ 42%] Built target absl_debugging_internal
[ 42%] Built target absl_base
[ 43%] Built target absl_throw_delegate
[ 44%] Building CXX object oead/lib/abseil/absl/strings/CMakeFiles/absl_strings_internal.dir/internal/ostringstream.cc.o
[ 44%] Building CXX object oead/lib/abseil/absl/strings/CMakeFiles/absl_strings_internal.dir/internal/utf8.cc.o
Scanning dependencies of target absl_bad_variant_access
Scanning dependencies of target absl_scoped_set_env
[ 44%] Building CXX object oead/lib/abseil/absl/types/CMakeFiles/absl_bad_variant_access.dir/bad_variant_access.cc.o
[ 45%] Built target absl_bad_optional_access
[ 46%] Building CXX object oead/lib/abseil/absl/base/CMakeFiles/absl_scoped_set_env.dir/internal/scoped_set_env.cc.o
<oead_path>/oead/lib/rapidyaml/src/c4/yml/tree.cpp:1453:15: error: no matching function for call to 'from_chars'
        if( ! from_chars(tk, &idx))
              ^~~~~~~~~~
<oead_path>/oead/lib/rapidyaml/ext/c4core/src/c4/charconv.hpp:943:1: note: candidate function not viable: no known conversion from 'size_t *' (aka 'unsigned long *') to 'void **__restrict' for 2nd argument
_C4_DEFINE_TO_FROM_CHARS(void*   , "p"             , "p"             )
^
<oead_path>/oead/lib/rapidyaml/ext/c4core/src/c4/charconv.hpp:926:13: note: expanded from macro '_C4_DEFINE_TO_FROM_CHARS'
inline bool from_chars(csubstr buf, ty * C4_RESTRICT v)                 \
            ^
<oead_path>/oead/lib/rapidyaml/ext/c4core/src/c4/charconv.hpp:944:1: note: candidate function not viable: no known conversion from 'size_t *' (aka 'unsigned long *') to 'float *__restrict' for 2nd argument
_C4_DEFINE_TO_FROM_CHARS_TOA(   float, f)
^
<oead_path>/oead/lib/rapidyaml/ext/c4core/src/c4/charconv.hpp:838:13: note: expanded from macro '_C4_DEFINE_TO_FROM_CHARS_TOA'
inline bool from_chars(csubstr buf, ty *C4_RESTRICT v)          \
            ^
<oead_path>/oead/lib/rapidyaml/ext/c4core/src/c4/charconv.hpp:945:1: note: candidate function not viable: no known conversion from 'size_t *' (aka 'unsigned long *') to 'double *__restrict' for 2nd argument
_C4_DEFINE_TO_FROM_CHARS_TOA(  double, d)
^
<oead_path>/oead/lib/rapidyaml/ext/c4core/src/c4/charconv.hpp:838:13: note: expanded from macro '_C4_DEFINE_TO_FROM_CHARS_TOA'
inline bool from_chars(csubstr buf, ty *C4_RESTRICT v)          \
            ^
<oead_path>/oead/lib/rapidyaml/ext/c4core/src/c4/charconv.hpp:946:1: note: candidate function not viable: no known conversion from 'size_t *' (aka 'unsigned long *') to 'int8_t *__restrict' (aka 'signed char *__restrict') for 2nd argument
_C4_DEFINE_TO_FROM_CHARS_TOA(  int8_t, i)
^
<oead_path>/oead/lib/rapidyaml/ext/c4core/src/c4/charconv.hpp:838:13:[ 47%] Building CXX object oead/lib/abseil/absl/strings/CMakeFiles/absl_strings_internal.dir/internal/escaping.cc.o
 note: expanded from macro '_C4_DEFINE_TO_FROM_CHARS_TOA'
inline bool from_chars(csubstr buf, ty *C4_RESTRICT v)          \
            ^
<oead_path>/oead/lib/rapidyaml/ext/c4core/src/c4/charconv.hpp:947:1: note: candidate function not viable: no known conversion from 'size_t *' (aka 'unsigned long *') to 'int16_t *__restrict' (aka 'short *__restrict') for 2nd argument
_C4_DEFINE_TO_FROM_CHARS_TOA( int16_t, i)
^
<oead_path>/oead/lib/rapidyaml/ext/c4core/src/c4/charconv.hpp:838:13: note: expanded from macro '_C4_DEFINE_TO_FROM_CHARS_TOA'
inline bool from_chars(csubstr buf, ty *C4_RESTRICT v)          \
            ^
<oead_path>/oead/lib/rapidyaml/ext/c4core/src/c4/charconv.hpp:948:1: note: candidate function not viable: no known conversion from 'size_t *' (aka 'unsigned long *') to 'int32_t *__restrict' (aka 'int *__restrict') for 2nd argument
_C4_DEFINE_TO_FROM_CHARS_TOA( int32_t, i)
^
<oead_path>/oead/lib/rapidyaml/ext/c4core/src/c4/charconv.hpp:838:13: note: expanded from macro '_C4_DEFINE_TO_FROM_CHARS_TOA'
inline bool from_chars(csubstr buf, ty *C4_RESTRICT v)          \
            ^
<oead_path>/oead/lib/rapidyaml/ext/c4core/src/c4/charconv.hpp:949:1: note: candidate function not viable: no known conversion from 'size_t *' (aka 'unsigned long *') to 'int64_t *__restrict' (aka 'long long *__restrict') for 2nd argument
_C4_DEFINE_TO_FROM_CHARS_TOA( int64_t, i)
^
<oead_path>/oead/lib/rapidyaml/ext/c4core/src/c4/charconv.hpp:838:13: note: expanded from macro '_C4_DEFINE_TO_FROM_CHARS_TOA'
inline bool from_chars(csubstr buf, ty *C4_RESTRICT v)          \
            ^
<oead_path>/oead/lib/rapidyaml/ext/c4core/src/c4/charconv.hpp:950:1: note: candidate function not viable: no known conversion from 'size_t *' (aka 'unsigned long *') to 'uint8_t *__restrict' (aka 'unsigned char *__restrict') for 2nd argument
_C4_DEFINE_TO_FROM_CHARS_TOA( uint8_t, u)
^
<oead_path>/oead/lib/rapidyaml/ext/c4core/src/c4/charconv.hpp:838:13: note: expanded from macro '_C4_DEFINE_TO_FROM_CHARS_TOA'
inline bool from_chars(csubstr buf, ty *C4_RESTRICT v)          \
            ^
<oead_path>/oead/lib/rapidyaml/ext/c4core/src/c4/charconv.hpp:951:1: note: candidate function not viable: no known conversion from 'size_t *' (aka 'unsigned long *') to 'uint16_t *__restrict' (aka 'unsigned short *__restrict') for 2nd argument
_C4_DEFINE_TO_FROM_CHARS_TOA(uint16_t, u)
^
<oead_path>/oead/lib/rapidyaml/ext/c4core/src/c4/charconv.hpp:838:13: note: expanded from macro '_C4_DEFINE_TO_FROM_CHARS_TOA'
inline bool from_chars(csubstr buf, ty *C4_RESTRICT v)          \
            ^
<oead_path>/oead/lib/rapidyaml/ext/c4core/src/c4/charconv.hpp:952:1: note: candidate function not viable: no known conversion from 'size_t *' (aka 'unsigned long *') to 'uint32_t *__restrict' (aka 'unsigned int *__restrict') for 2nd argument
_C4_DEFINE_TO_FROM_CHARS_TOA(uint32_t, u)
^
<oead_path>/oead/lib/rapidyaml/ext/c4core/src/c4/charconv.hpp:838:13: note: expanded from macro '_C4_DEFINE_TO_FROM_CHARS_TOA'
inline bool from_chars(csubstr buf, ty *C4_RESTRICT v)          \
            ^
<oead_path>/oead/lib/rapidyaml/ext/c4core/src/c4/charconv.hpp:953:1: note: candidate function not viable: no known conversion from 'size_t *' (aka 'unsigned long *') to 'uint64_t *__restrict' (aka 'unsigned long long *__restrict') for 2nd argument
_C4_DEFINE_TO_FROM_CHARS_TOA(uint64_t, u)
^
<oead_path>/oead/lib/rapidyaml/ext/c4core/src/c4/charconv.hpp:838:13: note: expanded from macro '_C4_DEFINE_TO_FROM_CHARS_TOA'
inline bool from_chars(csubstr buf, ty *C4_RESTRICT v)          \
            ^
<oead_path>/oead/lib/rapidyaml/ext/c4core/src/c4/charconv.hpp:984:13: note: candidate function not viable: no known conversion from 'size_t *' (aka 'unsigned long *') to 'bool *__restrict' for 2nd argument
inline bool from_chars(csubstr buf, bool * C4_RESTRICT v)
            ^
<oead_path>/oead/lib/rapidyaml/ext/c4core/src/c4/charconv.hpp:1015:13: note: candidate function not viable: no known conversion from 'size_t *' (aka 'unsigned long *') to 'char *__restrict' for 2nd argument
inline bool from_chars(csubstr buf, char * C4_RESTRICT v)
            ^
<oead_path>/oead/lib/rapidyaml/ext/c4core/src/c4/charconv.hpp:1044:13: note: candidate function not viable: no known conversion from 'size_t *' (aka 'unsigned long *') to 'c4::csubstr *__restrict' (aka 'basic_substring<const char> *__restrict') for 2nd argument
inline bool from_chars(csubstr buf, csubstr *C4_RESTRICT v)
            ^
<oead_path>/oead/lib/rapidyaml/ext/c4core/src/c4/charconv.hpp:1073:13: note: candidate function not viable: no known conversion from 'size_t *' (aka 'unsigned long *') to 'c4::substr *__restrict' (aka 'basic_substring<char> *__restrict') for 2nd argument
inline bool from_chars(csubstr buf, substr * C4_RESTRICT v)
            ^
1 error generated.
make[2]: *** [oead/lib/rapidyaml/CMakeFiles/ryml.dir/src/c4/yml/tree.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
Scanning dependencies of target absl_random_internal_randen
Scanning dependencies of target absl_bad_any_cast_impl
[ 47%] Building CXX object oead/lib/abseil/absl/random/CMakeFiles/absl_random_internal_randen.dir/internal/randen.cc.o
make[1]: *** [oead/lib/rapidyaml/CMakeFiles/ryml.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[ 47%] Building CXX object oead/lib/abseil/absl/types/CMakeFiles/absl_bad_any_cast_impl.dir/bad_any_cast.cc.o
[ 47%] Linking CXX static library libc4core.a
/Library/Developer/CommandLineTools/usr/bin/ranlib: file: libc4core.a(language.cpp.o) has no symbols
/Library/Developer/CommandLineTools/usr/bin/ranlib: file: libc4core.a(language.cpp.o) has no symbols
[ 50%] Built target c4core
[ 50%] Linking CXX static library libabsl_bad_variant_access.a
/Library/Developer/CommandLineTools/usr/bin/ranlib: file: libabsl_bad_variant_access.a(bad_variant_access.cc.o) has no symbols
[ 51%] Linking CXX static library libabsl_bad_any_cast_impl.a
/Library/Developer/CommandLineTools/usr/bin/ranlib: file: libabsl_bad_variant_access.a(bad_variant_access.cc.o) has no symbols
warning: /Library/Developer/CommandLineTools/usr/bin/ranlib: warning for library: libabsl_bad_variant_access.a the table of contents is empty (no object file members in the library define global symbols)
[ 51%] Built target absl_bad_variant_access
/Library/Developer/CommandLineTools/usr/bin/ranlib: file: libabsl_bad_any_cast_impl.a(bad_any_cast.cc.o) has no symbols
/Library/Developer/CommandLineTools/usr/bin/ranlib: file: libabsl_bad_any_cast_impl.a(bad_any_cast.cc.o) has no symbols
warning: /Library/Developer/CommandLineTools/usr/bin/ranlib: warning for library: libabsl_bad_any_cast_impl.a the table of contents is empty (no object file members in the library define global symbols)
[ 51%] Built target absl_bad_any_cast_impl
[ 51%] Linking CXX static library libabsl_strings_internal.a
[ 51%] Linking CXX static library libabsl_scoped_set_env.a
[ 52%] Built target absl_scoped_set_env
[ 52%] Linking CXX static library libabsl_random_internal_randen.a
[ 52%] Built target absl_strings_internal
[ 52%] Built target absl_random_internal_randen
make: *** [all] Error 2
/usr/local/lib/python3.8/site-packages/setuptools/dist.py:458: UserWarning: Normalizing 'v1.1.2+0.g254adc2.dirty' to '1.1.2+0.g254adc2.dirty'
  warnings.warn(tmpl.format(**locals()))
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "<oead_path>/oead/setup.py", line 64, in <module>
    setuptools.setup(
  File "/usr/local/lib/python3.8/site-packages/setuptools/__init__.py", line 165, in setup
    return distutils.core.setup(**attrs)
  File "/usr/local/Cellar/python@3.8/3.8.5/Frameworks/Python.framework/Versions/3.8/lib/python3.8/distutils/core.py", line 148, in setup
    dist.run_commands()
  File "/usr/local/Cellar/python@3.8/3.8.5/Frameworks/Python.framework/Versions/3.8/lib/python3.8/distutils/dist.py", line 966, in run_commands
    self.run_command(cmd)
  File "/usr/local/Cellar/python@3.8/3.8.5/Frameworks/Python.framework/Versions/3.8/lib/python3.8/distutils/dist.py", line 985, in run_command
    cmd_obj.run()
  File "/usr/local/lib/python3.8/site-packages/setuptools/command/develop.py", line 38, in run
    self.install_for_development()
  File "/usr/local/lib/python3.8/site-packages/setuptools/command/develop.py", line 140, in install_for_development
    self.run_command('build_ext')
  File "/usr/local/Cellar/python@3.8/3.8.5/Frameworks/Python.framework/Versions/3.8/lib/python3.8/distutils/cmd.py", line 313, in run_command
    self.distribution.run_command(command)
  File "/usr/local/Cellar/python@3.8/3.8.5/Frameworks/Python.framework/Versions/3.8/lib/python3.8/distutils/dist.py", line 985, in run_command
    cmd_obj.run()
  File "<oead_path>/oead/setup.py", line 29, in run
    self.build_extension(ext)
  File "<oead_path>/oead/setup.py", line 55, in build_extension
    subprocess.check_call([cmake_name, '--build', '.'] + build_args, cwd=self.build_temp)
  File "/usr/local/Cellar/python@3.8/3.8.5/Frameworks/Python.framework/Versions/3.8/lib/python3.8/subprocess.py", line 364, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['cmake', '--build', '.', '-j8', '--config', 'Release']' returned non-zero exit status 2.
----------------------------------------
ERROR: Command errored out with exit status 1: /usr/local/opt/python@3.8/bin/python3.8 -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'<oead_path>/oead/setup.py'"'"'; __file__='"'"'<oead_path>/oead/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' develop --no-deps Check the logs for full command output.
```

**Comment by leoetlino** [comment]
Try to apply this fix (but replace C4_IOS with C4_MACOS): https://github.com/biojppm/c4core/pull/9/files

Hopefully this is the last thing that needs to be fixed... Sorry for the hassle :/

**Comment by zardini123** [comment]
Wait so your saying go to those lines of code changed in the PR and replace the preprocessor directives with C4_MACOS?

Thank you for the help so far!  I really appreciate it!

**Comment by leoetlino** [comment]
Yeah basically add those lines to c4/charconv.hpp and c4/format.hpp, but replace C4_IOS with C4_MACOS

**Comment by zardini123** [comment]
Alright it got farther but now:

```
<oead_path>/oead/src/gsheet.cpp:258:24: error: no matching member function for call to 'Write'
                    writer.Write(item);
                    ~~~~~~~^~~~~
```

**Comment by leoetlino** [comment]
That should be fixed now! I've also added macOS 10.15 + Python 3.8 to the CI build targets to catch these kinds of silly issues. From now on, binary builds will be provided for the latest version of macOS (10.15 as of today). Since you are using 10.14, you will still need to build oead from source, though.

Thanks for the bug report!

**Comment by zardini123** [comment]
I see that GitHub Actions forces the usage of Catalina, which is odd.  Plenty of users and developers using macOS has opted to stay with Mojave (11.14) for a long while because Catalina (11.15) does not support 32 bit apps anymore.  If you were able to change the macOS version to be something like 11.13, it'd be forward compatible and available to those who are essentially stuck with 11.14.

Though regardless, why isn't building with 11.15 backwards compatible with 11.14?  Is there something about the pip wheel metadata or something that forces it to 11.15?  Would it be possible to tell it "compile and install on <11.15 Macs simply by ```pip install oead```"?

**Comment by leoetlino** [comment]
Should be fixed by #7. Wheels now target 10.14+ -- 10.15 is apparently just the default target version. I don't think I can target even older versions though, as GitHub's 10.15 image only has the 10.14 and 10.15 SDKs (and 10.13's Xcode is too old for proper C++17 support).

**Comment by zardini123** [comment]
> Making "pip install oead" install from source would require uploading the source for oead code + every library and every dependency recursively for every release. Seems easier to require users to install from source directly :P

Very interesting.  Thank you for the information!  I think it'd be crucial to provide the OS versions supported in the readme in section Getting Started.  It'd certainly help newcomers understand why its not installing for them.  Also, providing info on _how_ to build from source (like the information you originally gave me plus git modules initialize) would be amazing.

Thank you again for your help!
